elastic/kibana
# [Telemetry] collect number of visualization saved in the past 7, 30 and 90 days

## Summary
Fixes #57847

We want to also add to each visualization stats the number of visualizations saved the last 30 and the last 90 days.
Right now we collect the stats with the following format:
`"visualization_types": {
            "input_control_vis": {
              "total": 3,
              "spaces_min": 3,
              "spaces_max": 3,
              "spaces_avg": 3
            },
            "area": {
              "total": 4,
              "spaces_min": 4,
              "spaces_max": 4,
              "spaces_avg": 4
            },`

and with this change is going to also collect the number of visualizations for the past 30 and 90 days:
` "visualization_types": {
            "input_control_vis": {
              "total": 3,
              "spaces_min": 3,
              "spaces_max": 3,
              "spaces_avg": 3,
              "saved_30_days_total": 2,
              "saved_90_days_total": 3,
            },
            "area": {
              "total": 4,
              "spaces_min": 4,
              "spaces_max": 4,
              "spaces_avg": 4,
              "saved_30_days_total": 2,
              "saved_90_days_total": 4,
            },`

**How to test it:**
1. Go to Dev Tools and remove all the tasks
`POST /.kibana_task_manager/_delete_by_query
{
  "query": {
    "match_all": {}
  }
}`

2. Make a change to a server file in kibana (add an empty space for example)
3. Go to Management-->Advanced Setting->Telemetry-->See an example of what we collect

![Elastic](https://user-images.githubusercontent.com/17003240/83635782-750e0600-a5ad-11ea-9251-8f442e68b553.png)

Check the visualization types field, it should contain the new values.

### Checklist

- [x] This was checked for cross-browser compatibility, (https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
